src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1287,6 +1287,88 @@ public static bool TryInferParamsCollectionIterationType(Binder binder, TypeSymb
             return false;
         }
 
+#nullable enable
+        /// <returns>false if the set does not have a unique signature</returns>
+        internal bool FilterMethodsForUniqueSignature(ArrayBuilder<MethodSymbol> methods)
```

Comment:
>FilterMethodsForUniqueSignature [](http://example.com/codeflow?start=22&length=31)

I think the goal should be to reuse and share code with regular overload resolution process instead of writing new code that attempts to duplicate the logic. There is no other way to guarantee consistency in behavior, in my opinion.

Reply:
Sounds good, I will encapsulate the common logic in a private method once the implementation is final.

Reply:
> Sounds good, I will encapsulate the common logic in a private method once the implementation is final.

I would start with this, i.e. would not attempt to duplicate the logic. However, if you prefer to start with duplication, that would be fine. Please ping me once you ger rid of the duplication an I will take another look.

Reply:
As I mentioned during our offline conversation last week, I would try to split up ```PerformMemberOverloadResolution``` method and reuse its first part.